tests/component.js

```diff
@@ -0,0 +1,351 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Vue from 'vue';
+import { mount } from '@vue/test-utils';
+import CKEditorComponent from '../src/ckeditor';
+import sinon from 'sinon';
+import { getEditorNamespace } from '../src/utils/geteditornamespace';
+
+/* global window */
+const CKEDITOR = window.CKEDITOR;
+
+describe( 'CKEditor Component', () => {
+	const spies = {};
+
+	let skipReady = false;
+	let sandbox, wrapper, component, props;
+
+	beforeEach( done => {
+		spies.replace = sinon.spy( CKEDITOR, 'replace' );
+		spies.inline = sinon.spy( CKEDITOR, 'inline' );
+
+		wrapper = createComponent( props );
+		component = wrapper.vm;
+		sandbox = sinon.createSandbox();
+
+		if ( skipReady ) {
+			done();
+		} else {
+			component.$once( 'ready', () => {
+				done();
+			} );
+		}
+	} );
+
+	afterEach( () => {
+		skipReady = false;
+
+		for ( const key in spies ) {
```

Comment:
Do we really need names for spies? I'd use array for `spies` instead.

Reply:
Note how spies are used in this test file. They are referenced by name by multiple test steps and then restored after each test.